pkg/operator/controller/machineconfig/controller.go

```diff
@@ -37,6 +38,27 @@ import (
 //+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigpools,verbs=get;list;watch;create;delete
 //+kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;watch;patch
 
+// New returns a new MachineConfigReconciler instance.
+func New(mgr ctrlmgr.Manager, impls ...impl) *MachineConfigReconciler {
+	c := NewClient(impls...)
```

Comment:
I'm not sure if I'm in favor of this compared to a `impl: &defaultImpl{…}` and injecting it during testing by just setting `mgr.impl = …`